falcon/request.py

```diff
@@ -475,6 +480,59 @@ def headers(self):
     def params(self):
         return self._params
 
+    @property
+    def param(self):
+        if self._param is None:
+            self._param = helpers.ParamProxy(self, self.params)
+
+        return self._param
+
+    @property
+    def raw_body(self):
```

Comment:
That might do it. I just tested on gunicorn under 2.7 and 3.4 and stream.read() returns `bytes`, and `b''` when the stream is at EOF. wsgiref is similar. You may want to check mod_wsgi and uWSGI as well.